### PR TITLE
CHI-387 - Implement the “View” [ ] button to display a case.

### PR DESCRIPTION
**Pull request by andvirga**
Jira related: 
https://bugs.benetech.org/browse/CHI-387

Primary reviewer: @GPaoloni 

Changes description:
- Added logic in the reducer to make sure that when a case is updated, it is updated across all tasks.
- Added logic to update the case in All Cases View.
- Added some types.

### CODE REVIEW

File: plugin-hrm-form/src/components/caseList/CaseList.tsx [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+}
+
+const standaloneTask: StandaloneITask = {
+  taskSid: 'standalone-task-sid',
+  attributes: { isContactlessTask: false },
+};
+
+type OwnProps = {
+  setConnectedCase: (currentCase: CaseType, taskSid: string, caseHasBeenEdited: Boolean) => void;
+};
```
Comment (GPaoloni) [inline plugin-hrm-form/src/components/caseList/CaseList.tsx:76]: If you are typing the component like
`type Props = OwnProps & ConnectedProps<typeof connector>;`
you don't need to specify this on `OwnProps`, as `setConnectedCase` will be pulled out from `mapDispatchToProps`'s return (that's what `ConnectedProps<typeof connector>` does). So I think here you can leave `OwnProps = {}`

**Review by GPaoloni — APPROVED** [review]
Works great!
One thing I've spotted: If you have a case open in task search, then go to standalone search (or all cases) and update that same case, the old version of it is still open in task search (vice-versa is also true). I don't know if we should update it, or if "closing all open cases that matches with the updated one" is a better approach.

File: plugin-hrm-form/src/components/caseList/CaseList.tsx [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+// eslint-disable-next-line no-use-before-define
+type Props = OwnProps & ConnectedProps<typeof connector>;
+
+const CaseList: React.FC<Props> = ({ setConnectedCase }) => {
+  // const [showCaseDetails, setDisplayCase] = useState(false);
```
Comment (murilovmachado) [inline plugin-hrm-form/src/components/caseList/CaseList.tsx:82]: Looks like you can just delete this comment.

**Review by murilovmachado — APPROVED** [review]
Looks very good. I only have two comments:
1. This CaseList (All Cases) component doesn't keep its state when navigating to StandaloneSearch or to the main App. It's not related to this PR, so don't worry about it, but it's one little inconsistency in our App.
2. I wouldn't worry about trying to keep everything about Cases in sync, but since you've already implemented it, this is good the way you did it. But there are other harder scenarios that will continue to exist, such as two different counselours editing the same case at the same time. (No need to worry about it)

**Pull request merged**